app/server/lib/ActiveDoc.ts

```diff
@@ -972,20 +972,19 @@ export class ActiveDoc extends EventEmitter {
     }
     const attachments = this.docData.getMetaTable('_grist_Attachments').getRecords();
     const attachmentFileManager = this._attachmentFileManager;
+    const doc = this;
 
     async function* fileGenerator(): AsyncGenerator<ArchiveEntry> {
-      const filesSeen = new Set<string>();
       for (const attachment of attachments) {
-        if (filesSeen.has(attachment.fileIdent)) {
-          continue;
+        if (doc._shuttingDown) {
+          throw new ApiError("Document is shutting down, archiving aborted", 500);
         }
```

Comment:
I don't really like this approach, because there's not technically an ActiveDoc dependency here - there's a DocStorage dependency, which happens to be shut down when ActiveDoc is.  I'd like us to refactor shutdown handling at some point across ActiveDoc and all dependencies, but this feels okay for now?